text/0000-language-semver.md

```diff
@@ -0,0 +1,512 @@
+  [#24278]). In some cases, changes here may be backwards compatible,
+  or may be more appropriate only with explicit opt-in (or perhaps an
+  alternate macro system altogether, such as [this proposal][macro]).
+- Lints will evolve over time (both the lints that are enabled and the
+  precise cases that lints catch). We expect to introduce a
+  [means to limit the effect of these changes on dependencies][#1029].
+- Stack overflow is currently detected via a segmented stack check
+  prologue and results in an abort. We expect to experiment with a
+  system based on guard pages in the future.
+- We currently abort the process on OOM conditions (exceeding the heap space, overflowing
+  the stack). We may attempt to panic in such cases instead if possible.
+- Some details of type inference may change. For example, we expect to
+  implement the fallback mechanism described in [RFC 213], and we may
+  wish to make minor changes to accommodate overloaded integer
+  literals. In some cases, type inferences changes may be better
+  handled via explicit opt-in.
+
+There are other kinds of changes that can be made in a minor version
+that may break unsafe code but which are not considered breaking
+changes, because the unsafe code is relying on things known to be
+intentionally unspecified. One obvious example is the layout of data
+structures, which is considered undefined unless they have a
+`#[repr(C)]` attribute.
+
+Although it is not directly covered by this RFC, it's worth noting in
+passing that some of the CLI flags to the compiler may change in the
+future as well. The `-Z` flags are of course explicitly unstable, but
+some of the `-C`, rustdoc, and linker-specific flags are expected to
+evolve over time (see e.g. [#24451]).
+
+### Opt-in changes
+
+For breaking changes that are not related to soundness or language
+semantics, but are still deemed desirable, an opt-in strategy can be
+used instead. This section describes an attribute for opting in to
+newer language updates, and gives guidelines on what kinds of changes
+should or should not be introduced in this fashion.
+
+We use the term *"opt-in changes"* to refer to changes that would be
+breaking changes, but are not because of the opt-in mechanism.
+
+#### Rust version option
+
+The specific proposal is to introduce a command-line option
+`--rust-version=X.Y[.Z]` that instructs the Rust compiler to expect
+source code from older versions of Rust. This option could also be
+specified in a `Cargo.toml` file in a `rust-version` property.  The
+version applies to the crate currently being compiled and is called
+the crate's "supplied version". Every build of the Rust compiler will
+also have a version number built into it reflecting the current
+release; if the command-line option is not supplied, the compiler
+defaults to this builtin version.
+
+The supplied version is used by the compiler to produce the semantics
+of Rust "as it was" during version `X.Y`. RFCs that propose opt-in
```

Comment:
Hm, I'm not sure I understand this opt-in discussion. Is the hypothetical scenario building a crate that uses an opt-in feature from rust 1.2, but passing, say, `--rust-version=1.1`? If so, I don't see how this makes sense (why would one be opting-in to a version of Rust that can't handle the stuff the code uses?).